pkg/apiserver/audit/audit.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	"bufio"
+	"net"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+
+	"k8s.io/kubernetes/pkg/api"
+	utilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+// auditResponseWriter implements http.ResponseWriter interface.
+type auditResponseWriter struct {
+	http.ResponseWriter
+	id string
+}
+
+func (a *auditResponseWriter) WriteHeader(code int) {
+	glog.Infof("AUDIT: id=%q response=\"%d\"", a.id, code)
```

Comment:
A separate log file is not uncommon (depends on whether you want your logs
in one place or two places) - putting this behind an interface certainly
can't hurt though, and fleshing out the implementation.

On Wed, Jun 8, 2016 at 6:56 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/apiserver/audit/audit.go
> <https://github.com/kubernetes/kubernetes/pull/27087#discussion_r66355213>
> :
>
> > +
> > +	"github.com/golang/glog"
> > +	"github.com/pborman/uuid"
> > +
> > +	"k8s.io/kubernetes/pkg/api"
> > +	utilnet "k8s.io/kubernetes/pkg/util/net"
> > +)
> > +
> > +// auditResponseWriter implements http.ResponseWriter interface.
> > +type auditResponseWriter struct {
> > +	http.ResponseWriter
> > +	id string
> > +}
> > +
> > +func (a *auditResponseWriter) WriteHeader(code int) {
> > +	glog.Infof("AUDIT: id=%q response=\"%d\"", a.id, code)
>
> Is writing to stdout really the best way to produce audit logs? I guess
> I'd expect a separate output file, though I've never made audit logs before.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27087/files/752c551c39be25f16e03d37bec8a3c2567467370#r66355213>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p4gePInwAQlDTfd0K-tGyId2SyZeks5qJ0iqgaJpZM4Ixcs->
> .
>